pkg/serdes/serdes_test.go

```diff
@@ -666,6 +790,78 @@ func TestProtobufSerdesNestedValid(t *testing.T) {
 	req.NoError(os.RemoveAll(dir))
 }
 
+func TestProtobufSerdesValidWithRuleSet(t *testing.T) {
```

Comment:
@rayokota This is the test function showing deserializer decryption fails to transform data.